### PR TITLE
added MoU upload functionality

### PR DESCRIPTION
**Pull request by Rajsnehil**
Targets #432 

## Summary
This PR includes:-
- adding multiple MoU upload functionality. 
- We can upload pdf, and Docx-type files in the MoU section of the institution module.
- defined MoU in the specific section
- added start and date for MoU
- made the MoU URL link for previewing
- some code formatting

## Test Plan

- [x] Wrote tests
- [x] Tested locally

### CODE REVIEW

File: src/views/Institutions/Institution.jsx [hunk 1]
```diff
@@ -57,36 +62,79 @@ const Institute = (props) => {
...
+        console.log("UPDATE_DETAILS_ERR", err);
+        setAlert("Unable to update institution.", "error");
+      })
+      .finally(() => {
+        NP.done();
       getThisInstitution();
-    });
-    setModalShow(false);
+      });
+          setModalShow(false);
```
Comment (rathorevaibhav) [inline src/views/Institutions/Institution.jsx:120]: please fix indentation

File: src/views/Institutions/Institution.jsx [hunk 2]
```diff
@@ -57,36 +62,79 @@ const Institute = (props) => {
...
+              data: {
+                data: {
+                    upload: { id: uploadedMouId }
+                  }
+              }
+            } = await uploadFile(mouData.mou)
+
+            dataToSave['mou_list'].push({
+              ...mouData,
+              mou: uploadedMouId
```
Comment (rathorevaibhav) [inline src/views/Institutions/Institution.jsx:94]: let's make reduce the code size
```suggestion
            const response = await uploadFile(mouData.mou)
            dataToSave['mou_list'].push({
              ...mouData,
              mou: response.data.data.upload.id
```

Comment (rathorevaibhav) [inline src/views/Institutions/Institution.jsx:94]: let's make reduce the code size
```suggestion
            const response = await uploadFile(mouData.mou)
            dataToSave['mou_list'].push({
              ...mouData,
              mou: response.data.data.upload.id
```

File: src/views/Institutions/Institution.jsx [hunk 3]
```diff
@@ -57,36 +62,79 @@ const Institute = (props) => {
...
+            })
+          } catch (err) {
+          console.log('mou upload err', err)
+          setAlert("Unable to upload MoU.", "error")
```
Comment (rathorevaibhav) [inline src/views/Institutions/Institution.jsx:98]: indentation. Please do a self-review of your PR and fix indentation at all places

File: src/views/Institutions/InstitutionComponents/InstitutionForm.jsx [hunk 4]
```diff
@@ -104,37 +127,43 @@ const InstitutionForm = (props) => {
...
 
   if (!props.contacts) {
     // create an empty contact if no contacts are present
-    initialValues['contacts'] = [];
+    initialValues["contacts"] = [];
+  }
+
+  if (!props.mou_list) {
+    // create an empty MoU if no MoUs are present
+    initialValues["mou_list"] = [];
```
Comment (rathorevaibhav) [inline src/views/Institutions/InstitutionComponents/InstitutionForm.jsx:166]: let's rename `mou_list` to `mou` everywhere and `mou` to `mou_file`

File: src/views/Institutions/InstitutionComponents/MoUs.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import { useMemo } from "react";
+import Table from "../../../components/content/Table";
+import { Anchor } from "../../../components/content/Utils";
+
+const MoUs = ({ mou_list }) => {
```
Comment (rathorevaibhav) [inline src/views/Institutions/InstitutionComponents/MoUs.jsx:5]:
```suggestion
const MoUs = ({ mou }) => {
```

File: src/views/Institutions/InstitutionComponents/InstitutionForm.jsx [hunk 6]
```diff
@@ -265,6 +298,82 @@ const InstitutionForm = (props) => {
                   </div>
                 </div>
               </Section>
+              <Section>
+                <h3 className="section-header">MoU</h3>
+                <FieldArray name="mou_list">
+                  {({ insert, remove, push }) => (
```
Comment (rathorevaibhav) [inline src/views/Institutions/InstitutionComponents/InstitutionForm.jsx:304]: where are you using insert?

File: src/views/Institutions/InstitutionComponents/InstitutionForm.jsx [hunk 7]
```diff
@@ -265,6 +298,82 @@ const InstitutionForm = (props) => {
...
+                                placeholder="End Date"
+                                control="datepicker"
+                                className="form-control"
+                                autoComplete="off"
+                              />
+                            </div>
+                            <div className="col-md-6 col-sm-12 mb-2">
+                              <Input
+                                control="file"
+                                name={`mou_list.${index}.mou`}
```
Comment (rathorevaibhav) [inline src/views/Institutions/InstitutionComponents/InstitutionForm.jsx:336]:
```suggestion
                                name={`mou_list.${index}.mou_file`}
```

File: src/views/Institutions/InstitutionComponents/InstitutionForm.jsx [hunk 8]
```diff
@@ -265,6 +298,82 @@ const InstitutionForm = (props) => {
...
+                        ))}
+                      <div className="mt-2">
+                        <button
+                          className="btn btn-primary btn-sm"
+                          type="button"
+                          onClick={() => {
+                            push({
+                              start_date: "",
+                              end_date: "",
+                              mou: "",
```
Comment (rathorevaibhav) [inline src/views/Institutions/InstitutionComponents/InstitutionForm.jsx:366]:
```suggestion
                              mou_file: "",
```

File: src/views/Institutions/Institutions.jsx [hunk 9]
```diff
@@ -162,48 +190,60 @@ const Institutions = (props) => {
...
-    let {show, ...dataToSave} = data;
-
+    let { show, ...dataToSave } = data;
     nProgress.start();
-    createInstitution(dataToSave).then(data => {
-      if (data.data.errors) {
-        setFormErrors(data.data.errors);
-      } else {
-        setAlert("Institution created successfully.", "success");
+    createInstitution(dataToSave)
```
Comment (rathorevaibhav) [inline src/views/Institutions/Institutions.jsx:195]: are we not handling Mou file upload in create institution?

File: src/views/Institutions/InstitutionComponents/InstitutionForm.jsx (hunk 8 above)
Comment (rathorevaibhav) [inline src/views/Institutions/InstitutionComponents/InstitutionForm.jsx:366]:
```suggestion
                              mou_file: "",
```

File: src/views/Institutions/InstitutionComponents/InstitutionForm.jsx [hunk 10]
```diff
@@ -265,6 +298,82 @@ const InstitutionForm = (props) => {
...
+                              <Input
+                                control="file"
+                                name={`mou_list.${index}.mou`}
+                                label="MoU"
+                                className="form-control"
+                                placeholder="MoU"
+                                accept=".pdf, .docx"
+                                onChange={(event) => {
+                                  setFieldValue(
+                                    `mou_list.${index}.mou`,
```
Comment (rathorevaibhav) [inline src/views/Institutions/InstitutionComponents/InstitutionForm.jsx:343]:
```suggestion
                                    `mou_list.${index}.mou_file`,
```

**Review by rathorevaibhav — CHANGES_REQUESTED** [review]

File: src/views/Institutions/Institutions.jsx (hunk 9 above)
Comment (rathorevaibhav) [reply src/views/Institutions/Institutions.jsx:217]: @Rajsnehil I think you missed responding to this comment. Let me know.

**Review by rathorevaibhav — COMMENTED** [review]
@Rajsnehil one comment is still unaddressed. Please let me know your response.

File: src/views/Institutions/Institutions.jsx (hunk 9 above)
Comment (Rajsnehil) [reply src/views/Institutions/Institutions.jsx:217]: @rathorevaibhav, yes we will need Mou while creating an institution.

**Review by rathorevaibhav — APPROVED** [review]

**Pull request merged**